apps/api/src/common/user-auth-provider.service.ts

```diff
@@ -0,0 +1,203 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { AuthProvider, User } from '@prisma/client'
+import { PrismaService } from '@/prisma/prisma.service'
+
+@Injectable()
+export class UserAuthProviderService {
+  private readonly logger = new Logger(UserAuthProviderService.name)
+
+  constructor(private readonly prisma: PrismaService) {}
+
+  /**
+   * Adds an auth provider to a user's authProviders array if not already present
+   * @param userId - The ID of the user to add the auth provider to
+   * @param authProvider - The auth provider to add
+   * @returns The updated user
+   */
+  async addAuthProvider(
+    userId: string,
+    authProvider: AuthProvider
+  ): Promise<User> {
+    this.logger.log(`Adding auth provider ${authProvider} to user ${userId}`)
+
+    // Get the user with raw query to access authProviders field
+    const userResult = await this.prisma.$queryRaw<User[]>`
+      SELECT * FROM "User" WHERE id = ${userId}
+    `
+
+    if (!userResult || userResult.length === 0) {
+      throw new Error(`User with ID ${userId} not found`)
+    }
+
+    const user = userResult[0]
+
+    // Get current auth providers, fallback to legacy field if new field is empty
+    let currentProviders: AuthProvider[] = []
+
+    try {
+      // Try to access the new authProviders field
+      currentProviders = (user as any).authProviders || []
+    } catch {
+      // If it doesn't exist yet, use empty array
+      currentProviders = []
+    }
+
+    // Migrate from legacy field if needed
+    if (currentProviders.length === 0 && user.authProvider) {
```

Comment:
We would be applying a manual db command to move all the `authProvider` data into `authProviders`. You can remove this check